pkg/actor/validate_version.go

```diff
@@ -0,0 +1,280 @@
+		if err != nil {
+			return err
+		}
+		containerImage = dbContainer.Image
+		if strings.EqualFold(cluster.GetAnnotationVersion(), calVersion) {
+			log.Info("No update on version annotation -> nothing changed")
+			return nil
+		}
+		if strings.EqualFold(cluster.GetAnnotationContainerImage(), containerImage) {
+			log.Info("No update on container image annotation -> nothing changed")
+			return nil
+		}
+		cluster.SetClusterVersion(calVersion)
+		cluster.SetAnnotationVersion(calVersion)
+		cluster.SetCrdbContainerImage(containerImage)
+		cluster.SetAnnotationContainerImage(containerImage)
+		if err := v.client.Update(ctx, cluster.Unwrap()); err != nil {
+			log.Error(err, "failed saving the annotations on version checker")
+		}
+	} else {
+		// after 2 minutes the pod enters in the completed state
+		// if the container it is running and the version was not retrieved
+		// for instance if the image is nginx and we want to get the crdb version from it case
+		return PermanentErr{Err: errors.New("job completed with version empty-container running but no version was retrieved")}
+	}
+	dp := metav1.DeletePropagationForeground
+
+	//delete the job only if we have managed to get the version and we do not have any errors
+	err = clientset.BatchV1().Jobs(cluster.Namespace()).Delete(ctx, job.Name, metav1.DeleteOptions{
+		GracePeriodSeconds: ptr.Int64(5),
+		PropagationPolicy:  &dp,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete the job")
+	}
+
+	// we force the saving of the status on the cluster and cancel the loop
+	fetcher := resource.NewKubeFetcher(ctx, cluster.Namespace(), v.client)
+
+	cr := resource.ClusterPlaceholder(cluster.Name())
+	if err := fetcher.Fetch(cr); err != nil {
+		log.Error(err, "failed to retrieve CrdbCluster resource")
+		return err
+	}
+	
+	refreshedCluster := resource.NewCluster(cr)
+	// //save annotations first
+	// refreshedCluster.SetAnnotationVersion(calVersion)
+	// refreshedCluster.SetAnnotationContainerImage(containerImage)
+	// if err := v.client.Update(ctx, refreshedCluster.Unwrap()); err != nil {
+	// 	log.Error(err, "failed saving the annotations on version checker")
+	// 	return err
+	// }
+	// save the status of the cluster
+	refreshedCluster.SetFalse(api.CrdbVersionNotChecked)
+	refreshedCluster.SetClusterVersion(calVersion)
+	refreshedCluster.SetCrdbContainerImage(containerImage)
+	if err := v.client.Status().Update(ctx, refreshedCluster.Unwrap()); err != nil {
+		log.Error(err, "failed saving cluster status on version checker")
+		return err
+	}
+	log.Info("completed version checker")
+	CancelLoop(ctx)
+	return nil
+}
+
+func isJobCompletedOrFailed(job *kbatch.Job) (bool, kbatch.JobConditionType) {
+	for _, c := range job.Status.Conditions {
+		if (c.Type == kbatch.JobComplete || c.Type == kbatch.JobFailed) && c.Status == corev1.ConditionTrue {
+			return true, c.Type
+		}
+	}
+	return false, ""
+}
+
+func IsPodReady(
```

Comment:
This function is misleading, it doesn't actually wait for the pod to report itself as ready. There's a helper in the k8s SDK to check the readiness of a pod. You could also use a field selector.

Reply:
ok... The field selector running is not working as expected unfortunately. I will try the  helper. @chrisseto  can you tell me the package please and the helper name?